ios/PolyPodApp/PodApi/PolyIn/PolyIn.swift

```diff
@@ -1,10 +1,10 @@
 import CoreData
 
 protocol PolyInProtocol {
-    func addQuads(quads: [ExtendedData], completionHandler: (Bool) -> Void)
+    func addQuads(quads: [ExtendedData], completionHandler: (Error?) -> Void)
     func matchQuads(matcher: ExtendedData, completionHandler: ([ExtendedData]?, Error?) -> Void)
     func selectQuads(matcher: ExtendedData, completionHandler: ([ExtendedData]?, Error?) -> Void)
-    func deleteQuads(quads: [ExtendedData], completionHandler: (Bool) -> Void)
+    func deleteQuads(quads: [ExtendedData], completionHandler: (Error?) -> Void)
     func hasQuads(quads: [ExtendedData], completionHandler: (Bool) -> Void) -> Void
```

Comment:
Just for coherence, shouldn't this have an error too?

Reply:
Or make it optional?

Reply:
It is an error, Isn't it? The function returns success or error via the completionHandler - this code was always a bit of async hell, but I figure it's not quite time to refactor it just yet.

Reply:
I was referring to the line right above, the one that has not changed... Maybe it was a bit confusing.

Reply:
The line above being `selectQuads`? Or `matchQuads`? They both have an `Error` -> But since they also return a value, there's two parameters (again, weird, but kinda common practice). The only one that doesn't return an error, but a boolean indicating success instead, is `hasQuads` (`polyIn.has`).